### DISCUSSION TITLE
Hotend temperature starts to plummet once G29 command is issued.

### DISCUSSION THREAD

**Discussion by marcosgeorgopoulos** [mriscoc/Ender3V2S1#497] [Q&A] [answered]
Hi,
I recently installed Professional Firmware on my Ender 3 v2. Version 2.1.3 20221002
I'm trying to print something in ABS. I was previously using Jyers and did not experience this issue.
I my G-Code creates a mesh before printing with the G29 command.
However, as soon as the mesh starts getting created I can see the temp in my hot end drop at a fast rate, like 1 degree per second. Its as if the HotEnd is being turned off?
Then it starts to print but the temperature is too low ( printing ABS+ )
I've tried printing via octoprint as directly off an SD-Card. Either way I have the same issue.
```
;FLAVOR:Marlin
;TIME:12770
;Filament used: 9.71544m
;Layer height: 0.2
;MINX:55.903
;MINY:53.91
;MINZ:0.2
;MAXX:164.097
;MAXY:166.074
;MAXZ:21
;Generated with Cura_SteamEngine 5.2.1
M140 S100
M105
M190 S100
M104 S240
M105
M109 S240
M82 ;absolute extrusion mode
M201 X500.00 Y500.00 Z100.00 E5000.00 ;Setup machine max acceleration
M203 X500.00 Y500.00 Z20.00 E50.00 ;Setup machine max feedrate
M204 P500.00 R1000.00 T500.00 ;Setup Print/Retract/Travel acceleration
M205 X8.00 Y8.00 Z0.40 E5.00 ;Setup Jerk
M220 S100 ;Reset Feedrate
M221 S100 ;Reset Flowrate
G92 E0 ; Reset Extruder
G28 ; Home all axes
G29 ; Create mesh <- TEMPERATURE SEEMS TO DROP HERE!!!
G1 Z2.0 F3000 ; Move Z Axis up little to prevent scratching of Heat Bed
G1 X0.1 Y20 Z0.3 F5000.0 ; Move to start position
G1 X0.1 Y200.0 Z0.3 F1500.0 E15 ; Draw the first line
G1 X0.4 Y200.0 Z0.3 F5000.0 ; Move to side a little
G1 X0.4 Y20 Z0.3 F1500.0 E30 ; Draw the second line
G92 E0 ; Reset Extruder
G1 Z2.0 F3000 ; Move Z Axis up little to prevent scratching of Heat Bed
G1 X5 Y20 Z0.3 F5000.0 ; Move over to prevent blob squish
G92 E0
G92 E0
G1 F2700 E-5
;LAYER_COUNT:105
;LAYER:0
M107
G0 F6000 X62.939 Y91.41 Z0.2
;TYPE:SKIRT
G1 F2700 E0
```

Many thanks

**Comment by mriscoc** [comment, marked as answer]
Currently the firmware follows this suggestion:

https://github.com/mriscoc/Ender3V2S1/blob/1c733afd3d533959f6822341fadfc77dc88a9e72/Marlin/Configuration.h#L1584-L1590

But I'm planning to remove the `PROBING_HEATERS_OFF` configuration flag because of some issues with beginners. I think that people that want better probe accuracy will have more expertise to enable `PROBING_HEATERS_OFF` and compile the firmware by themselves.
The problem is with the PWM of the heaters that can cause EMI due to crosstalk, the wires of the BLTouch are very close to the wires of the heaters, that's why they were disabled before of taking a measurement.

You can re-heat after G29 using the G-code [M109](https://marlinfw.org/docs/gcode/M109.html)

**Reply by marcosgeorgopoulos** [reply]
Thanks I tried M109 directly after the G29 and the printer does not wait.  

G29
M190

It just keeping on printing and I have to pause the print manually until temperature is reached


Do I need to specify the temperature as well? If so is there a way that it can determine the current target?

**Reply by mriscoc** [reply]
Check the link to the Marlin documentation, you are missing the arguments for the 109 G-code.

**Reply by marcosgeorgopoulos** [reply]
I did check the documentation. It is vague but implies to waits for the target temp.

AUTOTEMP: Disable autotemp, **wait for temp**
M109

However it is  not 'waiting'  So there is either an issue with the documentation or the firmware.

**Reply by marcosgeorgopoulos** [reply]
thanks for updating your reply with a more helpful response.  The documentation does imply it will wait until the target temp is reached  and if I don't want to wait use 104.  Anyway, I guess we will have to manually update our generated g-code in the time being.

As a suggestion, you might wish to update your documentation where mentioning the use of G29 in the G-Code samples to mention you are disabling the heaters.
